modules/custom/az_paragraphs/tests/src/Unit/AZVideoEmbedHelperTest.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+namespace Drupal\Tests\az_paragraphs\Unit;
+
+use Drupal\az_paragraphs\AZVideoEmbedHelper;
+use Drupal\Tests\UnitTestCase;
+
+/**
+ * @coversDefaultClass \Drupal\az_paragraphs\AZVideoEmbedHelper
+ *
+ * @ingroup az_paragraphs
+ *
+ * @group az_paragraphs
+ */
+class AZVideoEmbedHelperTest extends UnitTestCase {
+
+  /**
+   * @var \Drupal\az_paragraphs\AZVideoEmbedHelper
+   */
+  protected $videoEmbedHelper;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp(): void {
+    $this->videoEmbedHelper = new AZVideoEmbedHelper();
+  }
+
+  /**
+   * Tests parsing of YouTube video ID from URLs.
+   *
+   * @covers ::getYoutubeIdFromUrl
+   *
+   * @dataProvider ::providerYouTubeVideoData
```

Comment:
```suggestion
   * @dataProvider providerYouTubeVideoData
```
This invalid provider syntax is causing this PHPunit warning (My mistake.  Sorry.):
```
The data provider specified for Drupal\Tests\az_paragraphs\Unit\AZVideoEmbedHelperTest::testGetYoutubeIdFromUrl is invalid.
PHPUnit\Util\Exception: Class  does not exist
```